### PR TITLE
Adding command line example

### PR DESCRIPTION
**Pull request by jose-compu**
`USAGE:
    cli [FLAGS] <SECRET_KEY> <MESSAGE> [PROOF]`

### CODE REVIEW

File: examples/cli.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    if proof_given {
+        pi = hex::decode(&matches.value_of("PROOF").unwrap()).unwrap();
+    } else {
+        println!("{}", hex::encode(&pi));
+    }
+
+    let beta = vrf.verify(&public_key, &pi, &message);
+
+    if proof_given {
+
```
Comment (girazoki) [inline examples/cli.rs]:
```suggestion
```

File: examples/cli.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        (author: "Vixify Network")
+        (about: "Client to Verifiable Random Functions")
+        (@arg VERBOSE: -v --verbose "Log verbosely to stderr.  This command does not currently log anything, so this option currently has no affect.")
+        (@arg SECRET_KEY: +required {is_hex_ok} "Secret key to be used to print or validate proof" )
+        (@arg MESSAGE: +required "Message to be used to print or validate proof." )
+        (@arg PROOF: {is_hex_ok} "Optional VRF Proof to be validated. If missing, proof is printed for secret and message." )
+    )
+    .get_matches();
+
+
```
Comment (girazoki) [inline examples/cli.rs]:
```suggestion
```

File: examples/cli.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    ($name:ident, $expr:expr) => {
+        fn $name(obj: String) -> Result<(), String> {
+            $expr(&obj).map(drop).map_err(|x| format!("{}", x))
+        }
+    };
+}
+gen_validator!(is_hex_ok, hex::decode);
+
+fn main() {
+
```
Comment (girazoki) [inline examples/cli.rs]:
```suggestion
```

File: examples/cli.rs [hunk 4]
```diff
@@ -0,0 +1,75 @@
+//! # Client example
+//!
+//! This example shows a command line client usage of the `vrf-rs` crate:
+//!
+//! 1. Instantiate the `ECVRF` by specifying the `CipherSuite`
+//! 2. Generate a VRF proof by using the `prove()` function
+//! 3. (Optional) Convert the VRF proof to a hash (e.g. to be used as pseudo-random value)
+//! 4. Verify a VRF proof by using `verify()` function
+
+use vrf::openssl::{CipherSuite, ECVRF};
```
Comment (girazoki) [inline examples/cli.rs:11]:
```suggestion
use hex;
use vrf::openssl::{CipherSuite, ECVRF};
```

File: examples/cli.rs [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+use vrf::VRF;
+use hex;
```
Comment (girazoki) [inline examples/cli.rs]:
```suggestion
```

**Review by girazoki — CHANGES_REQUESTED** [review]

**Comment by jose-compu** [comment]
Changes accepted!

File: examples/cli.rs [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            Ok(beta) => {
+                println!("VRF proof is valid!");
+                assert_eq!(hash, beta);
+            }
+            Err(_e) => {
+                println!("VRF proof is not valid!");
+            }
+        }
+    }
+
```
Comment (girazoki) [inline examples/cli.rs]:
```suggestion
```

**Comment by jose-compu** [comment]
Changes accepted.

**Review by girazoki — APPROVED** [review]

**Comment by girazoki** [comment]
Hi @manfr3d and thanks for the contribution! Can you squash your commits into the latest one? We require to have a clean and verified history in the repo. Thanks in advance!

**Comment by jose-compu** [comment]
That works is you neeed to remove the last commit? Thanks.

**Comment by girazoki** [comment]
No, just join all the commits together into the latest

**Comment by jose-compu** [comment]
Works now?

**Comment by girazoki** [comment]
I squashed them for you @manfr3d. This PR is closed by a15f43744756fe254ad42d8ed0367595ea1b6b2d

**Pull request closed without merging**